OnBoardingCalculator/app/src/main/java/com/globant/onboardingcalculator/mvp/model/CalculatorModel.java

```diff
@@ -56,4 +68,33 @@ public String getSecondOperand() {
     public char getOperator() {
         return operator;
     }
-}
+
+    public void operate() {
+        Double result;
+        switch (operator) {
+            case OPERATOR_PLUS:
+                result = parseDouble(firstOperand) + parseDouble(secondOperand);
+                clearOperation();
+                setFirstOperand(String.valueOf(df.format(result)));
+                break;
+            case OPERATOR_SUBSTRACTION:
+                result = parseDouble(firstOperand) - parseDouble(secondOperand);
+                clearOperation();
+                setFirstOperand(String.valueOf(df.format(result)));
+                break;
+            case OPERATOR_MULTIPLY:
+                result = parseDouble(firstOperand) * parseDouble(secondOperand);
+                clearOperation();
+                setFirstOperand(String.valueOf(df.format(result)));
+                break;
+            case OPERATOR_DIVIDE:
+                if (parseDouble(secondOperand) != parseDouble(NUMBER_ZERO)) {
+                    result = parseDouble(firstOperand) / parseDouble(secondOperand);
```

Comment:
why do you do the operation on the model instead of sending the result from the presenter?

Reply:
Nico is right, this must be in the presenter

Reply:
Yes Nico, it was one of my doubts, but now with this comment i realize that i have to change that. Thanks!

Reply:
👍